core/templates/services/rte-helper-modal.controller.spec.ts

```diff
@@ -158,4 +159,71 @@ describe('RteHelperModalComponent', () => {
       flush();
     }));
   });
+  describe('when there are validation errors in link form control', function() {
+    var customizationArgSpecs = [{
+      url: 'oppia.org',
+      text: 'oppia',
+    }];
+
+    beforeEach(() => {
+      fixture = TestBed.createComponent(RteHelperModalComponent);
+      component = fixture.componentInstance;
+      component.componentId = 'link',
+      component.attrsCustomizationArgsDict = {
+        heading: 'This value is not default.'
+      };
+      component.customizationArgSpecs = customizationArgSpecs;
+    });
+
+    it('should disable save button and display error message',
+      fakeAsync(() => {
+        component.ngOnInit();
+        flush();
+        component.customizationArgsForm.value[0] = 'oppia.org';
+        component.customizationArgsForm.value[1] = 'oppia';
+        component.onCustomizationArgsFormChange(
+          component.customizationArgsForm.value);
+        expect(component.saveButtonIsDisabled).toBe(true);
+        expect(component.errorMessage).toBe(
+          'It seems like clicking on this link will lead the user to a ' +
+          'different URL than the text specifies. Please change the text.'
+        );
+        flush();
+      }));
+  });
+  describe('when there are validation errors in video form control',
+    function() {
+      var customizationArgSpecs = [{
+        name: 'heading',
```

Comment:
What is "heading"? That's not a property of a video and I don't understand what you're testing here.

Reply:
done